loadtest/loadtest.go

```diff
@@ -128,18 +149,34 @@ func (lt *LoadTester) removeUsers(numUsers int) (int, error) {
 	}
 
 	var wg sync.WaitGroup
+
 	wg.Add(numUsers)
 	// TODO: Add a way to make how a user is removed decidable from the upper layer (the user of this API),
 	// for example by passing a typed constant (e.g. random, first, last).
-	for i := 0; i < numUsers; i++ {
-		go func(i int) {
+	var stopped []int
+
+	for id := range lt.activeControllers {
```

Comment:
Just noting that by using a map and iterating over it we lose ordering and with it the current behaviour of removing the latest added user.

Reply:
Yeah, any reason we can't do this with 2 slices ? Seems like we are just transferring controllers from one slice to another.

Reply:
No specific reason actually. I like the randomness of maps. However, everyone would be happier with slices, so it's ok.

Reply:
Yes, this was a decision taken early on in the project to try to make things as deterministic as possible. Even the logic to only remove the last entry from the slice was intentional and discussed in a code review. Because the problem with the previous load test was that results from consecutive runs would be very random.